examples/nlp/text_classification_with_transformer.py

```diff
@@ -0,0 +1,184 @@
+        v = self.wv(x)  # (batch_size, seq_len, embed_dim)
+        q = self.separate_heads(
+            q, batch_size
+        )  # (batch_size, num_heads, seq_len, projection_dim)
+        k = self.separate_heads(
+            k, batch_size
+        )  # (batch_size, num_heads, seq_len, projection_dim)
+        v = self.separate_heads(
+            v, batch_size
+        )  # (batch_size, num_heads, seq_len, projection_dim)
+        attention, weights = self.attention(q, k, v)
+        attention = tf.transpose(
+            attention, perm=[0, 2, 1, 3]
+        )  # (batch_size, seq_len, num_heads, projection_dim)
+        concat_attention = tf.reshape(
+            attention, (batch_size, -1, self.embed_dim)
+        )  # (batch_size, seq_len, embed_dim)
+        output = self.combine_heads(
+            concat_attention
+        )  # (batch_size, seq_len, embed_dim)
+        return output
+
+
+"""
+## Implement transformer block as a layer
+"""
+
+
+class TransformerLayer(tf.keras.layers.Layer):
+    def __init__(self, embed_dim, num_heads, ff_dim, rate=0.1):
+        super(TransformerLayer, self).__init__()
+
+        self.att = MultiHeadSelfAttention(embed_dim, num_heads)
+        self.ffn = keras.Sequential(
+            [
+                keras.layers.Dense(ff_dim, activation="relu"),
+                keras.layers.Dense(embed_dim),
+            ]
+        )
+
+        self.layernorm1 = keras.layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = keras.layers.LayerNormalization(epsilon=1e-6)
+
+        self.dropout1 = tf.keras.layers.Dropout(rate)
+        self.dropout2 = tf.keras.layers.Dropout(rate)
+
+    def call(self, x, training):
+        attn_output = self.att(x)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(x + attn_output)
+
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        out2 = self.layernorm2(out1 + ffn_output)
+
+        return out2
+
+
+"""
+## Implement embedding layer
+
+Two seperate embedding layers, one for tokens, one for token index (positions).
+"""
+
+
+class EmbeddingLayer(keras.layers.Layer):
+    def __init__(self, maxlen, vocab_size, emded_dim):
+        super(EmbeddingLayer, self).__init__()
+        self.token_emb = keras.layers.Embedding(
+            input_dim=vocab_size, output_dim=emded_dim
+        )
+        self.pos_emb = keras.layers.Embedding(input_dim=maxlen, output_dim=emded_dim)
+
+    def call(self, x):
+        maxlen = tf.shape(x)[-1]
+        positions = tf.range(start=0, limit=maxlen, delta=1)
+        positions = self.pos_emb(positions)
+        x = self.token_emb(x)
+        return x + positions
+
+
+"""
+## Create classifier model using transformer layer
+
+Transformer layer outputs one vector for each time step of your input sequence. Here, we take the mean across all time steps and build a two layered feed forward network on top of it.
```

Comment:
Break up text lines so that they are no longer than 90 char. This helps with version control and code readability.